execution/executor/src/components/apply_chunk_output.rs

```diff
@@ -235,14 +229,7 @@ impl ApplyChunkOutput {
         let keeps_and_discards = transaction_outputs.iter().map(|t| t.status()).cloned();
         let retries = repeat(TransactionStatus::Retry).take(to_retry.len());
 
-        let status = if state_checkpoint_to_add.is_some() {
-            keeps_and_discards
-                .chain(once(TransactionStatus::Keep(ExecutionStatus::Success)))
-                .chain(retries)
-                .collect()
-        } else {
-            keeps_and_discards.chain(retries).collect()
-        };
+        let status = keeps_and_discards.chain(retries).collect();
```

Comment:
Why removing this? State checkpoint is still inserted here but not in status?

Reply:
as far as I can tell - status corresponds to input transactions (and why order of keeps and discards is kept)

for execution/commit, we use to_keep - which contains status itself.


this was the confusing part, we depended on status here being in the wrong place.

I've changed consensus/consensus-types/src/executed_block.rs, to append both the status and the new transaction at the same time, for consistency

Reply:
But the state checkpoint txn is still added below in line 241?

Reply:
yes, that's what I mean. to_keep here seems to me to be the list of transactions that actually get committed

statuses here is always looked from the perspective of input transactions, and is kept in that order (not in the order they will be committed). let me rename statuses to statuses_for_input_txns?

Reply:
Sounds good to me as long as the storage bosses are happy cc @msmouse @grao1991

Reply:
How epoch ending is handled?

Reply:
epoch ending (i.e. reconfiguration) doesn't add StateCheckpoint/BlockPrologue transaction (i.e. adding transaction is gated upon that), i.e.:

let state_checkpoint_to_add =
new_epoch_marker.map_or_else(|| append_state_checkpoint_to_block, |_| None);